tests/rbac/test_middleware.py

```diff
@@ -344,5 +344,105 @@ def test_principal_no_access(self):
             "role": {"read": [], "write": []},
             "policy": {"read": [], "write": []},
         }
-        access = IdentityHeaderMiddleware._get_access_for_user()
+        access = IdentityHeaderMiddleware._get_access_for_user("test_user", self.tenant)
+        self.assertEqual(expected, access)
+
+    def test_principal_with_access_no_res_defs(self):
+        """Test a user with definded access without any resource definitions."""
+        principal = Principal.objects.create(username="test_user")
+        group = Group.objects.create(name="group1")
+        group.principals.add(principal)
+        group.save()
+        role = Role.objects.create(name="role1")
+        perm = Permission.objects.create(permission="rbac:group:write")
+        access = Access.objects.create(permission=perm, role=role)
+        policy = Policy.objects.create(name="policy1", group=group)
+        policy.roles.add(role)
+        policy.save()
+        access = IdentityHeaderMiddleware._get_access_for_user("test_user", self.tenant)
+        expected = {
+            "group": {"read": ["*"], "write": ["*"]},
```

Comment:
This is probably just my level of unfamiliarity with the tests speaking, but as I read this it _looks_ like we're confirming that the user's access level for the given tenant includes read and write permissions? Would this restrict the given user from granting "rbac:group:write" to other users?

Reply:
Hmm, I don't think that directly limits it. I'll look back over things and see how best we can block that. Probably by limiting creation of roles/groups with the 'rbac' application type.

Reply:
Oh! That should be limited by the blocks on what apps can have custom roles created. Currently we don't allow creation of RBAC custom roles, so only the default/system/canned roles will allow access to RBAC. Right, @coderbydesign @astrozzc ?

Reply:
Right, currently we don't allow custom roles for RBAC. Canned roles could be assigned to user to have this rbac permissions. 
Hmm, but once a user is given "rbac:group:write", he would be able to assigned the canned role to someone else's group and give them this "rbac:group:write" [here](https://github.com/RedHatInsights/insights-rbac/blob/master/rbac/management/permissions/group_access.py#L45)